test/runtests.jl

```diff
@@ -0,0 +1,16 @@
+using ChainRules
+using ChainRulesTestUtils
+using Random
+using Test
+
+@testset "ChainRulesTestUtils.jl" begin
+    double(x) = 2x
+    test_scalar(double, 2)
```

Comment:
@oxinabox this is starting to get to the point where I'm running out of knowledge.

The three tests in here are failing, on

> r_res = rrule(f, x)

Where the `rule` was not defined. Not sure how to keep moving forward on this.

Reply:
We need to define the function and it's rules. Since it's a scalar to scalar function, we can define the rule like `@scalarrule(double(x), 2)`, then ` test_scalar` should pass